x/group/go.mod

```
@@ -0,0 +1,142 @@
+module cosmossdk.io/x/group
+
+go 1.21.1
```

```suggestion
go 1.21
```

Otherwise nix complains